test/connection.test.ts

```diff
@@ -523,6 +523,50 @@ describe('Connection', function () {
       assert.calledOnce(spy)
       assert.calledWith(spy, '58')
     })
+
+    it('should not send money from server in receiveOnly mode', async function () {
+      const spy = sinon.spy(this.serverPlugin, 'sendData')
+      const zeroAmountPacket = sinon.match(
+        (data: Buffer) => IlpPacket.deserializeIlpPrepare(data).amount === '0'
+      )
+      spy.alwaysCalledWithExactly(zeroAmountPacket)
+
+      this.server = await createServer({
+        plugin: this.serverPlugin,
+        serverSecret: Buffer.alloc(32),
+        receiveOnly: true
+      })
+
+      const { destinationAccount, sharedSecret } = this.server.generateAddressAndSecret()
+      const connectionPromise = this.server.acceptConnection()
+
+      this.clientConn = await createConnection({
+        plugin: this.clientPlugin,
+        destinationAccount,
+        sharedSecret
+      })
+
+      await connectionPromise
+    })
+
+    it('should not send money from client in receiveOnly mode', async function ()  {
```

Comment:
Same as above, this is more about establishing the connection without sending test money packets then it just not being able to send money.